docs/explanations/design.rst

```diff
@@ -0,0 +1,26 @@
+.. SPDX-FileCopyrightText: 2023 The meson-python developers
+..
+.. SPDX-License-Identifier: MIT
+
+.. _explanations-design:
+
+*******************************
+How does ``meson-python`` work?
+*******************************
+
+
+.. todo::
+
+   - Explain how ``meson-python`` maps files to the wheel
+
+     - Link to :ref:`tutorials-executable` in the relevant section
+
+   - Explain why the heuristics are necessary and how they work
```

Comment:
Aren't we about to remove those heuristics? Or are these not the ones for `{moduledir_shared}` & co?

Reply:
We are, this was written before that PR.